WcaOnRails/app/webpacker/components/RolesTab/DelegateForm.jsx

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import { Form } from 'semantic-ui-react';
+import I18n from '../../lib/i18n';
+
+const delegateStatusOptions = [{
+  text: I18n.t('enums.user.delegate_status.trainee_delegate'),
+  value: 'trainee_delegate',
+}, {
+  text: I18n.t('enums.user.delegate_status.candidate_delegate'),
+  value: 'candidate_delegate',
+}, {
+  text: I18n.t('enums.user.delegate_status.delegate'),
+  value: 'delegate',
+}, {
+  text: I18n.t('enums.user.delegate_status.senior_delegate'),
+  value: 'senior_delegate',
+}];
+
+const defaultValue = {
+  delegateStatus: delegateStatusOptions[0].value,
+  location: '',
+};
+
+export default function DelegateForm({
+  formValues,
+  setFormValues,
```

Comment:
Ideally, this would take something like `updateFormProperty` which you could use like
```
onChange={(_, { value }) => updateFormProperty({ location: value })}
```
and not have to worry about maintaining the other properties (or knowing about the data structure).

Or even three separate props `updateDelegateStatus`, `updateSeniorDelegateId`, and `upfateLocation` instead. But that might be overkill.

But this is a pretty small, simple component, so it's fine as it is.

Reply:
That's interesting idea, I did that. Thanks Kevin.